models/news.models.js

```diff
@@ -0,0 +1,61 @@
+const db = require("../db/connection");
+const {articleAddComments} = require('../db/seeds/utils')
+
+exports.selectEndpoints = () => {
+    return new Promise((resolve, reject) => {
+        resolve(require("../endpoints.json"));
+    }).then((endpointData) => {
+        return endpointData;
+    }).catch((err)=>{
+        if (err.code === 'MODULE_NOT_FOUND'){
+            return Promise.reject({msg: 'Not Found'});
+        } else {
+            return Promise.reject(err);
+        }
+    });
+};
+
+exports.selectTopics = () => {
+    return db.query(`
+        SELECT * FROM topics;
+    `).then((topics)=>{
+        return topics.rows;
+    }).catch((err)=>{
+        console.log('CATCH ERROR')
+    });
+};
+
+exports.selectArticles = () => {
+    return db.query(`
+    SELECT * FROM articles
+    ORDER BY created_at DESC;
+    `).then((articles)=>{
+        const commentedArticles = articles.rows.map((article)=>{
+            return articleAddComments(article);
+        });
+
+        return Promise.all(commentedArticles)
```

Comment:
You shouldn't need to to map this - realistically, you can do this all via d.query. Have a look at COUNT SQL functions  :)